imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,159 @@
+  const currentUserName = getCurrentUserName(contextUserAccount); // TODO: make sure this works as it should
+
+  // check to see if invited user has an account
+  const invitedUser = await users.findOne({ "emails.address": email });
+
+  // if there is a userAccount, check to see if email has been verified
+  const matchingEmail = invitedUser && invitedUser.emails && invitedUser.emails.find((accountEmail) => accountEmail.address === email);
+  const isEmailVerified = matchingEmail && matchingEmail.verified;
+
+  // set variables to pass to email templates
+  let dataForEmail;
+  let userId;
+  let templateName;
+
+  if (invitedUser) {
+    userId = invitedUser._id;
+  }
+
+  // if the user already has an account, send in informative email instead of an invite email
+  if (invitedUser && isEmailVerified) {
+    // make sure user has an `Account` - this should always be the case
+    const invitedAccount = await Accounts.findOne({ _id: userId });
+    if (!invitedAccount) throw new ReactionError("not-found", "User found but matching account not found");
+
+    await context.mutations.addAccountToGroup({ ...context, isInternalCall: true }, {
+      accountId: invitedAccount._id,
+      groupId
+    });
+
+    // do not send token, as no password reset is needed
+    const url = Reaction.absoluteUrl();
+
+    // use primaryShop's data (name, address etc) in email copy sent to new shop manager
+    dataForEmail = getDataForEmail({ shop: primaryShop, currentUserName, name, url });
+
+    // Get email template and subject
+    templateName = "accounts/inviteShopMember";
+  } else { // TODO: this whole else section uses meteor
+    // There could be an existing user with an invite still pending (not activated).
+    // We create a new account only if there's no pending invite.
+    if (!invitedUser) {
+      // The user does not already exist, we need to create a new account
+      userId = MeteorAccounts.createUser({ // TODO: something with this since it's meteor
+        profile: { invited: true },
+        email,
+        name,
+        groupId
+      });
+    }
+
+    const token = Random.id();
+
+    // set token to be used for first login for the new account
+    const tokenUpdate = {
+      "services.password.reset": { token, email, when: new Date() },
+      name
+    };
+    Meteor.users.update(userId, { $set: tokenUpdate }); // TODO: something with this since it's meteor
+
+    // use primaryShop's data (name, address etc) in email copy sent to new shop manager
+    dataForEmail = getDataForEmail({ shop: primaryShop, currentUserName, name, token });
+
+    // Get email template and subject
+    templateName = "accounts/inviteNewShopMember";
+  }
+
+  dataForEmail.groupName = _.startCase(group.name);
+
+  // send invitation email from primary shop email
+  await context.mutations.sendEmail(context, {
+    data: dataForEmail,
+    fromShop: primaryShop,
+    templateName,
+    to: email
+  });
+
+  return Accounts.findOne({ userId });
```

Comment:
@aldeed the old version of this meteor method did not have any `appEvent` at the end, should there be something here?

Reply:
The `MeteorAccounts.createUser` call causes `Accounts.onCreateUser` to run, and that emits "afterAccountCreate", so I think we're okay as is.